multiverse/tipmanager.go

```diff
@@ -90,37 +115,38 @@ func (t *TipManager) Tips() (strongTips MessageIDs, weakTips MessageIDs) {
 	tipSet := t.TipSet(t.tangle.OpinionManager.Opinion())
 
 	strongTips = tipSet.StrongTips(config.TipsCount, t.tsa)
-	weakTips = tipSet.WeakTips(config.TipsCount-1, t.tsa)
+	// In the paper we consider all strong tips
+	// weakTips = tipSet.WeakTips(config.TipsCount-1, t.tsa)
 
-	// if no weak tips is available, then return whatever number of strong tips
-	if len(weakTips) == 0 {
-		return
-	}
+	// Remove the weakTips-related codes
+	// if len(weakTips) == 0 {
+	// 	return
+	// }
 
-	if strongTipsCount := len(strongTips); strongTipsCount < OptimalStrongTipsCount {
-		fillUpCount := config.TipsCount - strongTipsCount
+	// if strongTipsCount := len(strongTips); strongTipsCount < OptimalStrongTipsCount {
+	// 	fillUpCount := config.TipsCount - strongTipsCount
 
-		if fillUpCount >= len(weakTips) {
-			return
-		}
+	// 	if fillUpCount >= len(weakTips) {
+	// 		return
+	// 	}
 
-		weakTips.Trim(fillUpCount)
-		return
-	}
+	// 	weakTips.Trim(fillUpCount)
+	// 	return
+	// }
 
-	if weakTipsCount := len(weakTips); weakTipsCount < OptimalWeakTipsCount {
-		fillUpCount := config.TipsCount - weakTipsCount
+	// if weakTipsCount := len(weakTips); weakTipsCount < OptimalWeakTipsCount {
+	// 	fillUpCount := config.TipsCount - weakTipsCount
 
-		if fillUpCount >= len(strongTips) {
-			return
-		}
+	// 	if fillUpCount >= len(strongTips) {
+	// 		return
+	// 	}
 
-		strongTips.Trim(fillUpCount)
-		return
-	}
+	// 	strongTips.Trim(fillUpCount)
+	// 	return
+	// }
 
 	strongTips.Trim(OptimalStrongTipsCount)
```

Comment:
This should stay commented out if we do not consider weak tips, OptimalStrongTipsCount was just a helper variable to determine the proportion between stron and weak tips.